From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Mon, 5 Feb 2018 21:33:08 +0100
Subject: [PATCH] {MAYBE UPS} x86: mce: silence 64-bit-int-to-32-bit-pointer
 cast warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

x86_32 gcc reports:

    arch/x86/kernel/cpu/mcheck/mce.c: In function ‘__print_mce’:
    arch/x86/kernel/cpu/mcheck/mce.c:237:21: error: cast to pointer from
    integer of different size [-Werror=int-to-pointer-cast]
        pr_cont("{%pS}", (void *)m->ip);
                         ^
    include/linux/printk.h:315:26: note: in definition of macro ‘pr_cont’
      printk(KERN_CONT fmt, ##__VA_ARGS__)
                              ^~~~~~~~~~~
---
 arch/x86/kernel/cpu/mcheck/mce.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/kernel/cpu/mcheck/mce.c b/arch/x86/kernel/cpu/mcheck/mce.c
index d494f60a5235..e31ac431e182 100644
--- a/arch/x86/kernel/cpu/mcheck/mce.c
+++ b/arch/x86/kernel/cpu/mcheck/mce.c
@@ -234,7 +234,7 @@ static void __print_mce(struct mce *m)
 			m->cs, m->ip);
 
 		if (m->cs == __KERNEL_CS)
-			pr_cont("{%pS}", (void *)m->ip);
+			pr_cont("{%pS}", (void *)(uintptr_t)m->ip);
 		pr_cont("\n");
 	}
 
-- 
